src/Raven.Studio/typescript/models/database/tasks/replicationAccessBaseModel.ts

```diff
@@ -0,0 +1,114 @@
+    
+    hubToSinkPrefixes = ko.observableArray<prefixPathModel>([]);
+    sinkToHubPrefixes = ko.observableArray<prefixPathModel>([]);
+    
+    samePrefixesForBothDirections = ko.observable<boolean>(false);
+    filteringPathsRequired = ko.observable<boolean>(true);
+        
+    dirtyFlag = new ko.DirtyFlag([]);
+
+    constructor(accessName: string, certificate: replicationCertificateModel, hubToSink: prefixPathModel[], sinkToHub: prefixPathModel[], filteringPathsRequired: boolean = true) {
+       
+        this.replicationAccessName(accessName);
+        this.hubToSinkPrefixes(hubToSink);
+        this.sinkToHubPrefixes(sinkToHub);
+        this.certificate(certificate);
+      
+        this.samePrefixesForBothDirections(_.isEqual(hubToSink.map(x => x.path()), sinkToHub.map(x => x.path())));
+       
+        this.filteringPathsRequired(filteringPathsRequired);
+    }
+    
+    initObservables() {
+        this.dirtyFlag = new ko.DirtyFlag([
+            this.replicationAccessName,
+            this.certificate,
+            this.hubToSinkPrefixes,
+            this.sinkToHubPrefixes,
+            this.samePrefixesForBothDirections
+        ], false, jsonUtil.newLineNormalizingHashFunction);
+    }
+    
+    initValidation() {
+        this.replicationAccessName.extend({
+           required: true
+        });
+        
+        this.certificate.extend({
+            required: true
+        })
+        
+        this.hubToSinkPrefixes.extend({
+            validation: [
+                {
+                    validator: () => !this.filteringPathsRequired() || this.hubToSinkPrefixes().length,
+                    message: "Please add at least one filtering prefix path."
+                }
+            ]
+        })
+
+        this.sinkToHubPrefixes.extend({
+            validation: [
+                {
+                    validator: () => !this.filteringPathsRequired() || this.samePrefixesForBothDirections() || this.sinkToHubPrefixes().length,
+                    message: "Please add at least one filtering prefix path, or use the Hub to Sink prefixes."
+                }
+            ]
+        })
+    }
+
+    addHubToSinkInputPrefixWithBlink() {
+        if (!this.hubToSinkPrefixes().filter(prefix => prefix.path() === this.inputPrefixHubToSink().path()).length) {
+            const itemToAdd = new prefixPathModel(this.inputPrefixHubToSink().path());
+            this.hubToSinkPrefixes().unshift(itemToAdd);
+            this.hubToSinkPrefixes(this.hubToSinkPrefixes());
+
+            this.inputPrefixHubToSink().path(null);
+            $("#hubToSink .collection-list li").first().addClass("blink-style");
+        }
+    }
+
+    addSinkToHubInputPrefixWithBlink() {
+        if (!this.sinkToHubPrefixes().filter(prefix => prefix.path() === this.inputPrefixSinkToHub().path()).length) {
+            const itemToAdd = new prefixPathModel(this.inputPrefixSinkToHub().path());
+            this.sinkToHubPrefixes().unshift(itemToAdd);
```

Comment:
why unshift on array not on observable array ?

Reply:
done